implementations/rust/ockam/ockam_api/src/cli_state/test_support.rs

```diff
@@ -1,12 +1,30 @@
 use crate::cli_state::Result;
 use crate::cli_state::{random_name, CliState, CliStateError};
+use ockam_node::database::SqlxDatabase;
 use std::path::PathBuf;
 
 /// Test support
 impl CliState {
     /// Return a test CliState with a random root directory
+    /// Use this CliState for a simple integration test since every call to that function deletes
+    /// all previous state if the database being used is Postgres.
     pub async fn test() -> Result<Self> {
-        Self::create(Self::test_dir()?).await
+        let test_dir = Self::test_dir()?;
+
+        // clean the existing state if any
+        let db = SqlxDatabase::create(CliState::make_database_configuration(&test_dir)?).await?;
+        db.drop_all_postgres_tables().await?;
```

Comment:
Should we hide implementation details here?

Reply:
Maybe but so far I don't really know what's the right abstraction to introduce there.